src/Traits/BelongsToThrough.php

```diff
@@ -11,13 +11,16 @@ trait BelongsToThrough
     /**
      * Define a belongs-to-through relationship.
      *
-     * @param string $related
-     * @param array|string $through
+     * @template TRelatedModel of \Illuminate\Database\Eloquent\Model
+     * @template TIntermediateModel of \Illuminate\Database\Eloquent\Model
+     *
+     * @param class-string<TRelatedModel> $related
+     * @param class-string<TIntermediateModel>[]|array{0: class-string<TIntermediateModel>, 1: string}[]|class-string<TIntermediateModel> $through
```

Comment:
This only doesn't take into account the class-string as alias, so PHPStan can't infer that usage. I think it's not technically possible to infer that